### DISCUSSION TITLE
Keep Low Frequency Data When Converting I2S 24 bit to 16 bit

### DISCUSSION THREAD

**Discussion by pauldupuyjr** [pschatzmann/arduino-audio-tools#1825] [Q&A] [answered]
I need the lowest frequency audio from a 24 bit ICS-43434 mic. I'm using Equilizer3Bands.freq_low and gain settings to isolate the band, as well as Boost in an AudioEffectStream to increase volume. Because these require 16-bit input, I've got the I2SStream bits_per_sample configuration set to 16. 

All of this seems to be working; however, I'm not getting the expected output. Research suggests this may be due to the 24 to 16 bit conversion truncating the lowest frequency data. If that's the case, it seems like a good solution would be changing the conversion to drop the higher values rather than lower. 

I've been poking around in I2SDriverESP32 and NumberFormatConverterStream and am not sure how to go about this. I'm hoping there's a built-in way to do what I want, as has turned out to be the case for everything else in this incredible library. ;-) Thanks in advance for advice or pointers.

**Comment by pschatzmann** [comment, marked as answer]
- Use the VolumeStream to amplify the signal: there is a chapter in the Wiki, so read this first.
- The bits per example should not impact the frequency range range at all: it is the sampling frequency that defines the available frequency range, but in my understanding a higher sample rate just increases the upper available frequencies (to half the sampling rate), so I suggest to try with 16 bits first and if you are insterested only in lower frequencies try to keep the sample rate as low as possible!

**Comment by pauldupuyjr** [comment]
Thanks for the quick reply Phil! I'm using 16 bits now (so I can use the awesome equilizer!). I am concerned about the result of the "native" 24 bit I2S mic data being converted when received from the I2S channel. Some data must be lost? Is the cast dropping the lower or higher bits to fit into the 16 bit number? This discussion suggests it is the lower bits: "The quietest parts of a digital recording (like the very end of a long reverb tail) use bits 1-8. The middle of the line signal fall somewhere between 9-16. The loud stuff is up there at 17-24 bits. When you truncate to from 24 to 16 bits, you are effectively throwing away bits 1-8." https://recording.org/comment/mjones4th/24-bit-16-bit-whats-lost/lets-simple-it-expansion-gaffs-statement-clo#comment-18371

**Reply by pschatzmann** [reply]
Hmm, that might be theoretically correct, I would not be sure if this is something that you would be able to realistically measure (with your hardware). 

You can try to do some experiments with 24 or 32 bits....

**Reply by pauldupuyjr** [reply]
If I could use the Equilizer3Bands with 24 bit input, that would be ideal. Is there a plan to add this capability?

**Reply by pschatzmann** [reply]
I double checked the code yesterday and it should work with 16, 24 and 32 bits
